lib/jwt/decode.rb

```diff
@@ -113,6 +113,8 @@ def segment_length
     end
 
     def none_algorithm?
+      return false unless algorithm.respond_to? :casecmp
```

Comment:
I wonder if it would just be simpler to check if it's a string or just is truthy.

```
return false unless algorithm
```

```
return false unless algithm.is_a?(String)
```

Reply:
Maybe even the original approach would be the simplest.

`algorithm == 'none'`

Reply:
Checking for a specific type is not very Ruby-way approach. Because Ruby promotes duck typing, and therefore it shouldn't matter if a class is directly based on a `String` or it implements exactly the same interface independently - the code should work anyways. So checking for the presence of a certain method is quite common in Ruby to ensure the interface of an object is supported, I think it is intentionally preferred over checking the actual class.

Checking `algorithm == 'none'` is fine, the only thing is that it stops supporting case-insensitive matches, but I have quickly checked the RFC and it never explicitly requires to support anything other than `none`.

I have also found that algorithm name is supposed to be case sensitive. 

![image](https://user-images.githubusercontent.com/10187586/179391741-de4749a2-a427-4cde-adb2-2c850a9a6910.png)

And `none` is listed as an algorithm name:

![image](https://user-images.githubusercontent.com/10187586/179391837-00554ab8-1c95-4eb6-b6b1-070245ba131f.png)

 So I guess supporting something like `None` or `nOnE` could be seen as a spec violation.

https://datatracker.ietf.org/doc/html/rfc7518#section-7.1.1